File: string/117.cpp
#include <bits/stdc++.h>

using namespace std;

bool isPrefixString(string s, vector<string>& words) {
    int pos = 0;
    
    for(int i = 0; i < words.size(); ++i) {
        int res = s.find(words[i], pos);
        if(res == string::npos || res != pos)
            return false;

        pos += words[i].size();
        if (pos >= s.size())
            return true;
    }

    if(pos < s.size())
        return false;

    return true;
}

int main() {
    string s = "ccccccccc";
    vector<string> words = {"c","cc"};
    cout<<boolalpha<<isPrefixString(s, words);

    return 0;
}

File: string/117.py
from typing import List


def isPrefixString(s: str, words: List[str]) -> bool:
    pos = 0

    for index in range(len(words)):
        found = s.find(words[index], pos)
        if found == -1 or found != pos:
            return False

        pos += len(words[index])
        if pos >= len(s):
            return True

    if pos < len(s):
        return False

    return True


print(isPrefixString("cccccccc", ["c", "cc"]))
